Rollbar/RollbarQueueController.cs

```diff
@@ -214,6 +219,14 @@ private void ProcessQueues(AccessTokenQueuesMetadata tokenMetadata)
             }
         }
 
+        private void ObeyPayloadTimeout(Payload payload, PayloadQueue queue)
+        {
+            if (payload.TimeoutAt.HasValue && (DateTime.Now.Add(this._sleepInterval) >= payload.TimeoutAt.Value))
+            {
+                queue.Dequeue();
```

Comment:
Doesn't this mean that we would not signal the signal inside the payload?

Reply:
we signal the semaphore only in case of successful transmission of the payload to the rollbar api service, otherwise it just times out and gets disposed of in either case at the end of the blocking method.